certbot/tests/renewal_test.py

```diff
@@ -54,6 +54,26 @@ def test_webroot_params_conservation(self, mock_set_by_cli):
         self.assertEqual(self.config.webroot_map, {})
         self.assertEqual(self.config.webroot_path, ['/var/www/test'])
 
+    def test_reuse_key_renewal_params(self):
+        self.config.rsa_key_size = 'INVALID_VALUE'
+        self.config.reuse_key = True
+        self.config.dry_run = True
+        config = configuration.NamespaceConfig(self.config)
+
+        rc_path = test_util.make_lineage(
+            self.config.config_dir, 'sample-renewal.conf')
+        lineage = storage.RenewableCert(rc_path, config)
+
+        le_client = mock.MagicMock()
+        le_client.obtain_certificate.return_value = (None, None, None, None)
+
+        from certbot._internal import renewal
+
+        with mock.patch('certbot._internal.renewal.hooks.renew_hook'):
+            renewal.renew_cert(self.config, None, le_client, lineage)
+
+        assert self.config.rsa_key_size == 2048
```

Comment:
This is already the default value. Perhaps it'd make more sense to test with a different key size?

Reply:
Just to clarify the purpose of this assertion here. The point is not to verify that the key is actually of 2048 bits (which is the default indeed), but to check that the option for the rsa key size has been set in the configuration, so it will be stored in the renewal params later.

Indeed you will see that I set this value to `INVALID_VALUE` and expect it to be set to the actual size of the key during the renewal call.